src/model/deepdives.py

```diff
@@ -48,28 +59,71 @@ class Warning(str, Enum):
     SHIELD_DISRUPTION = "Shield Disruption"
     SWARMAGEDDON = "Swarmageddon"
 
+class Mission():
+    """ Represents one mission objective like collect 7 Aquarq or mine 200 Morkite"""
+    name: str
+
+    def __init__(self, name: str):
+        self.name = name
+
+    @property
+    def emoji(self) -> str:
+        return get_emoji(self.type)
+
+    @property
+    def type(self) -> MissionType:
+        """Mission types provided by API don't have much overlap with regular
+        mission types thus, there needs to be such conversion"""
+
+        type_name_matrix: dict[MissionType] = {
+            'morkite': MissionType.MINING_EXPEDITION,
+            'egg':  MissionType.EGG_HUNT,
+            'refining': MissionType.ON_SITE_REFINING,
+            'mule': MissionType.SALVAGE_OPERATION,
+            'aquarq': MissionType.POINT_EXTRACTION,
+            'escort':  MissionType.ESCORT_DUTY,
+            'dreadnought': MissionType.ELIMINATION,
+            'black box':  MissionType.BLACK_BOX,
+            'sabotage':  MissionType.INDUSTRIAL_SABOTAGE
+        }
+
+        for (name, type) in type_name_matrix.items():
+            if name in self.name.lower():
+                return type
```

Comment:
Instead of adding this somewhat redundant mission type attribute in order to map the emoji, could you perhaps use a regular expression to determine which emoji to use for the pre-existing mission name?

Reply:
I think that regex is not needed and unnecessary complexity here. Code that uses just `name in self.name.lower()` is much easier to understand.